cobalt/dom/media_source.cc

```diff
@@ -406,11 +427,31 @@ bool MediaSource::IsTypeSupported(script::EnvironmentSettings* settings,
                                   const std::string& type) {
   TRACE_EVENT1("cobalt::dom", "MediaSource::IsTypeSupported()", "type", type);
   DCHECK(settings);
-  DOMSettings* dom_settings =
-      base::polymorphic_downcast<DOMSettings*>(settings);
-  DCHECK(dom_settings->can_play_type_handler());
-  SbMediaSupportType support_type =
-      dom_settings->can_play_type_handler()->CanPlayAdaptive(type.c_str(), "");
+
+  web::EnvironmentSettings* web_settings =
+      base::polymorphic_downcast<web::EnvironmentSettings*>(settings);
+  DCHECK(web_settings);
+  DCHECK(web_settings->context());
+  web::WindowOrWorkerGlobalScope* global_scope =
+      web_settings->context()->GetWindowOrWorkerGlobalScope();
+
+  SbMediaSupportType support_type;
+  if (global_scope->IsWorker()) {
```

Comment:
I saw `IsWorker()` can be `IsDedicatedWorker()` or `IsServiceWorker()`. Does the use case of this only in `DedicatedWorker`? If yes, should this be `IsDedicatedWorker()`?